src/ui/control/geolocate_control.js

```diff
@@ -325,6 +342,21 @@ class GeolocateControl extends Evented {
         }
     }
 
+    /**
+     * Update the user location dot Marker rotation to the current heading
+     *
+     * @private
+     */
+    _updateMarkerRotation() {
+        if (this._userLocationDotMarker && typeof this._heading === 'number') {
+            this._userLocationDotMarker.setRotation(this._heading);
+            this._dotElement.classList.add('mapboxgl-user-location-show-heading');
+        } else {
+            this._dotElement.classList.remove('mapboxgl-user-location-show-heading');
```

Comment:
Should we move manipulating `classList` in the update method from here to the `trigger` method which hosts all other lines like this?

Reply:
@mourner I thought about that. The trigger would add an event listener for `devicrorientation` but it should not show the heading until the information is available in `_updateMarkerRotation`.